numpy/tests/test_numpy.py

```diff
@@ -0,0 +1,149 @@
+    assert arr3.sum(2) == np.array([[[[1.5, 2.5, 3.5], [3.5, 4.5, 5.5], [5.5, 6.5, 7.5]]]])
+    assert arr3.sum(3) == np.array([[[[10.5, 13.5, 16.5]]]])
+    assert arr3.sum(4) == np.array([[[[7.5, 13.5, 19.5]]]])
+    assert arr3.sum((0, 1)) == np.array([[[1.5, 2.5, 3.5], [3.5, 4.5, 5.5], [5.5, 6.5, 7.5]]])
+    assert arr3.sum((3, 4)) == np.array([[[40.5]]])
+
+def test_array_prod():
+    arr1 = np.array([1, 2, 3, 4, 5, 6, 7, 8, 9])
+    assert arr1.prod() == 362880
+    assert arr1.prod(0) == np.array(362880)
+
+    arr2 = np.array([[1], [2], [3]])
+    assert arr2.prod() == 6
+    assert arr2.prod(0) == np.array([6])
+    assert arr2.prod(1) == np.array([1, 2, 3])
+    assert arr2.prod((0,1)) == np.array(6)
+
+    arr3 = np.array([[[[[1.5, 2.5, 3.5], [3.5, 4.5, 5.5], [5.5, 6.5, 7.5]]]]])
+    assert arr3.prod() == 304845.556640625
+    assert arr3.prod(0) == np.array([[[[1.5, 2.5, 3.5], [3.5, 4.5, 5.5], [5.5, 6.5, 7.5]]]])
+    assert arr3.prod(1) == np.array([[[[1.5, 2.5, 3.5], [3.5, 4.5, 5.5], [5.5, 6.5, 7.5]]]])
+    assert arr3.prod(4) == np.array([[[[13.125,   86.625,  268.125]]]])
+    assert arr3.prod((0, 3)) == np.array([[[28.875,   73.125,  144.375]]])
+
+def test_array_min():
+    arr1 = np.array([1, 2, 3, 4, 5, 6, 7, 8, 9])
+    assert arr1.min() == 1
+
+    arr2 = np.array([[-1], [-2], [-3]])
+    assert arr2.min() == -3
+
+    arr3 = np.array([[[[[1.5, -1.5], [3.5, -3.5], [5.5, -5.5]]]]])
+    assert arr3.min() == -5.5
+    assert arr3.min(0) == np.array([[[[1.5, -1.5], [3.5, -3.5], [5.5, -5.5]]]])
+    assert arr3.min(4) == np.array([[[[-1.5, -3.5, -5.5]]]])
+    assert arr3.min((0, 3)) == np.array([[[1.5, -5.5]]])
+    assert arr3.min((0, 4)) == np.array([[[-1.5, -3.5, -5.5]]])
+
+def test_array_max():
+    arr1 = np.array([1, 2, 3, 4, 5, 6, 7, 8, 9])
+    assert arr1.max() == 9
+
+    arr2 = np.array([[-1], [-2], [-3]])
+    assert arr2.max() == -1
+
+    arr3 = np.array([[[[[1.5, -1.5], [3.5, -3.5], [5.5, -5.5]]]]])
+    assert arr3.max() == 5.5
+    assert arr3.max(0) == np.array([[[[1.5, -1.5], [3.5, -3.5], [5.5, -5.5]]]])
+    assert arr3.max(4) == np.array([[[[1.5, 3.5, 5.5]]]])
+    assert arr3.max((0, 3)) == np.array([[[5.5, -1.5]]])
+    assert arr3.max((0, 4)) == np.array([[[1.5, 3.5, 5.5]]])
+
+def test_array_mean():
+    arr = np.array([[[1.5, 2.5, 3.5], [3.5, 4.5, 5.5], [5.5, 6.5, 7.5]]])
+    assert arr.mean() == 4.5
+    assert arr.mean(0) == np.array([[1.5, 2.5, 3.5], [3.5, 4.5, 5.5], [5.5, 6.5, 7.5]])
+    assert arr.mean(1) == np.array([[3.5, 4.5, 5.5]])
+    assert arr.mean(2) == np.array([[2.5, 4.5, 6.5]])
+
+def test_array_std():
+    arr = np.array([[[1.5, 2.5, 3.5], [3.5, 4.5, 5.5], [5.5, 6.5, 7.5]]])
+    assert arr.std() == 1.8257418583505538
+    assert arr.std(0) == np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0]])
+    assert repr(arr.std(1)) == repr(np.array([[1.632993,  1.632993,  1.632993]]))
+    assert repr(arr.std(2)) == repr(np.array([[ 0.816497,  0.816497,  0.816497]]))
```

Comment:
@16bit-ykiko Do you have something better in mind here ? I guess due to some truncation, values of high floating point precision give false on equation (but obviously print the same).

Reply:
Have you implemented `numpy.allclose`? I think you can use it to specifie tolerance.

Reply:
I was thinking of having that. Thanks for the validation.